docs/tutorials/forum-application/6.main.md

```diff
@@ -0,0 +1,200 @@
+# Main function
+
+**This main function is responsible for running the Forum Application blockchain.**
+
+```go
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	app "github.com/andynog/abci2-forum-app/abci"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/cometbft/cometbft/p2p"
+	"github.com/cometbft/cometbft/proxy"
+	"github.com/spf13/viper"
+
+	db "github.com/cometbft/cometbft-db"
+	cfg "github.com/cometbft/cometbft/config"
+	cmtflags "github.com/cometbft/cometbft/libs/cli/flags"
+	cmtlog "github.com/cometbft/cometbft/libs/log"
+	nm "github.com/cometbft/cometbft/node"
+	"github.com/cometbft/cometbft/privval"
+)
+
+var homeDir string
+
+func init() {
+	flag.StringVar(&homeDir, "cmt-home", "", "Path to the CometBFT config directory (if empty, uses $HOME/.cometbft)")
+}
+
+func main() {
+	flag.Parse()
+	if homeDir == "" {
+		homeDir = os.ExpandEnv("$HOME/.cometbft")
+	}
+
+	config := cfg.DefaultConfig()
+	config.SetRoot(homeDir)
+	viper.SetConfigFile(fmt.Sprintf("%s/%s", homeDir, "config.toml"))
+
+	if err := viper.ReadInConfig(); err != nil {
+		log.Fatalf("failed to read config: %v", err)
+	}
+
+	db, err := db.NewGoLevelDB(filepath.Join(homeDir, "forum-db"), ".")
+	if err != nil {
+		log.Fatalf("failed to create database: %v", err)
+	}
+	defer db.Close()
+
+	dbPath := "forum-db"
+	appConfigPath := "app.toml"
+	app, err := app.NewForumApp(dbPath, appConfigPath)
+
+	if err != nil {
+		log.Fatalf("failed to create ForumApp instance: %v", err)
+	}
+
+	logger := cmtlog.NewTMLogger(cmtlog.NewSyncWriter(os.Stdout))
+	logger, err = cmtflags.ParseLogLevel(config.LogLevel, logger, cfg.DefaultLogLevel)
+
+	if err != nil {
+		log.Fatalf("failed to read genesis doc: %v", err)
+	}
+
+	nodeKey, err := p2p.LoadNodeKey(config.NodeKeyFile())
+	if err != nil {
+		log.Fatalf("failed to load node key: %v", err)
+	}
+
+	pv := privval.LoadFilePV(
+		config.PrivValidatorKeyFile(),
+		config.PrivValidatorStateFile(),
+	)
+
+	node, err := nm.NewNode(
+		context.Background(),
+		config,
+		pv,
+		nodeKey,
+		proxy.NewLocalClientCreator(app),
+		nm.DefaultGenesisDocProviderFunc(config),
+		cfg.DefaultDBProvider,
+		nm.DefaultMetricsProvider(config.Instrumentation),
+		logger,
+	)
+
+	if err != nil {
+		log.Fatalf("failed to create CometBFT node: %v", err)
+	}
+
+	if err := node.Start(); err != nil {
+		log.Fatalf("failed to start CometBFT node: %v", err)
+	}
+	defer func() {
+		node.Stop()
+		node.Wait()
+	}()
+
+	httpAddr := "127.0.0.1:8080"
+	http.HandleFunc("/messages", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
What is this for?

Reply:
just realized that this endpoint is not doing anything properly. I've removed it for now, thanks.